GitHubReferences.ps1

```diff
@@ -0,0 +1,525 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+function Get-GitHubReference
+{
+<#
+    .SYNOPSIS
+        Retrieve a reference from a given GitHub repository.
+
+    .DESCRIPTION
+        Retrieve a reference from a given GitHub repository.
+        The Git repo for this module can be found here: http://aka.ms/PowerShellForGitHub
+
+    .PARAMETER OwnerName
+        Owner of the repository.
+        If not supplied here, the DefaultOwnerName configuration property value will be used.
+
+    .PARAMETER RepositoryName
+        Name of the repository.
+        If not supplied here, the DefaultRepositoryName configuration property value will be used.
+
+    .PARAMETER Uri
+        Uri for the repository.
+        The OwnerName and RepositoryName will be extracted from here instead of needing to provide
+        them individually.
+
+    .PARAMETER Tag
+        The name of the Tag to be retrieved.
+
+    .PARAMETER Branch
+        The name of the Branch to be retrieved.
+
+    .PARAMETER All
+        If provided, will retrieve both branches and tags for the given repository
+
+    .PARAMETER MatchPrefix

```

Comment:
This is the option for calling the matching-refs API. Wasn't sure what is the standard pattern for such a functionality